src/main/java/pl/edu/mimuw/usos/Usos.java

```diff
@@ -0,0 +1,139 @@
+package pl.edu.mimuw.usos;
+
+import pl.edu.mimuw.usos.Faculty.*;
+import pl.edu.mimuw.usos.People.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Usos {
+  private final String name;
+  private final static IndexCounter indexStudentsCounter = new IndexCounter();
```

Comment:
zauważ że przez to że te zmienna zadeklarowałeś jako statyczna, w klasie `IndexCounter`, będzie ona wspólna dla tych wszystkich 3 counterów - chyba nie o to Ci chodziło i wtedy po prostu zmienna `indexCounter` w tamtej klasie nie powinna być statyczna